src/util/element-object-generator.ts

```diff
@@ -40,4 +41,35 @@ export namespace ElementObjectGenerator {
                 return null;
         }
     };
+
+    export const duplicate = (element: TElement, existingIds: string[]) => {
+        let internalId: string;
+        while (!internalId || existingIds.includes(internalId)) {
+            internalId = SimpleIdGenerator.generate();
+        }
+
+        const generateId = (id: string) => {
+            switch (true) {
+                case id.includes("-copy-"): {
+                    const parts = id.split("-");
+                    const number = parseInt(parts[parts.length - 1], 10);
+                    const incrementedNumber = number + 1;
+                    const newId =
+                        parts.slice(0, -1).join("-") + "-" + incrementedNumber;
+                    return newId;
+                }
+                case id.includes("-copy"):
+                    return id + "-2";
+                default:
+                    return id + "-copy";
+            }
```

Comment:
it should also be checking if `<id>-copy` and `<id>-copy-<number>` exists in `existingIds`

else if I duplicate `emailAddress` twice, I will have 2 instances of `emailAddress-copy`

Reply:
issue still remains, to replicate:
1. add email address element
2. assign an id, e.g. `email` and save it
3. duplicate element: `email-copy` is added correctly
4. duplicate the original `email` element again, `email-copy` is added instead of `email-copy-2`

when an element is duplicated, there is a need to:
- extract the underlying id, e.g. extract `email` out of `email-copy`
- check against all the existing elements to determine how many `email` and `email-copy` are there
- generate the correct unique id accordingly

Reply:
these changes are in the latest push